lib/slack-ruby-bot/commands/base.rb

```diff
@@ -82,6 +79,17 @@ def self.finalize_routes!
         return if self.routes && self.routes.any?
         command default_command_name
       end
+
+      def self.get_gif_and_send!(client, channel, text, keywords, options = {})
```

Comment:
I think you're misusing the `!` method, it's usually for things that raise exceptions. If for example we weren't ignoring the result from `Giphy.random`, then it would make sense.

Maybe a better name is `send_message_with_gif_for_keywords`? 

Furthermore since I am nitpicking to death here, this is an internal method and gives us a lot of flexibility in terms of parameters. A method with 5 parameters is 3 too many :) 

I would do this:

```ruby
def self.send_message_with_gif_for_keywords!(options = {})
   client = options[:client]
   channel = options[:channel]
   ...
end
```

And rewrite all the other methods consequently. If I were to redo this whole thing I would have probably make `send_message_with_gif` just take a hash too, but that ship has sailed and I don't think it's worth changing the interface.